domain/model/state/types.go

```diff
@@ -87,3 +91,84 @@ func (m *dbModel) toCoreModel() (coremodel.Model, error) {
 type dbModelUUID struct {
 	ModelUUID string `db:"model_uuid"`
 }
+
+// dbModelSummary stores the information from the model table for a model
+// summary.
+type dbModelSummary struct {
+	// Name is the model name.
+	Name string `db:"name"`
+	// UUID is the model unique identifier.
+	UUID string `db:"uuid"`
+	// Type is the model type (e.g. IAAS or CAAS).
+	Type string `db:"model_type"`
+	// OwnerName is the tag of the user that owns the model.
+	Owner string `db:"owner_name"`
+	// Life is the current lifecycle state of the model.
+	Life string `db:"life"`
+
+	// CloudName is the name of the model cloud.
+	CloudName string `db:"cloud_name"`
+	// Cloud type is the models cloud type.
+	CloudType string `db:"cloud_type"`
+	// CloudRegion is the region of the model cloud.
+	CloudRegion string `db:"cloud_region_name"`
+
+	// CloudCredentialName is the name of the cloud credential.
+	CloudCredentialName string `db:"cloud_credential_name"`
+	// CloudCredentialCloudName is the name of the cloud the credential is for.
+	CloudCredentialCloudName string `db:"cloud_credential_cloud_name"`
+	// CloudCredentialOwnerName is the name of the cloud credential owner.
+	CloudCredentialOwnerName string `db:"cloud_credential_owner_name"`
+
+	// Access is the access level the supplied user has on this model
+	Access permission.Access `db:"access_type"`
+	// UserLastConnection is the last time this user has accessed this model
+	UserLastConnection *time.Time `db:"time"`
+
+	// AgentVersion is the agent version for this model.
+	AgentVersion string `db:"target_agent_version"`
+}
+
+func (m dbModelSummary) toModelSummary(controllerInfo dbController) (model.ModelSummary, error) {
```

Comment:
nit: I'm not really a fan of `toX` type of methods, because they're not symmetrical with `fromX`, or you have to try really hard to make them. Generally, I prefer just calling a method; `encodeModelSummary` and `decodeModelSummary`.

Reply:
Changed the name to decodeModelSummary. Though in this case there will never be an encodeModelSummary since this data is not put in to the DB as a summary.